modules/ingestor/src/service/advisory/csaf/loader.rs

```diff
@@ -36,43 +36,46 @@ impl<'g> CsafLoader<'g> {
             .ingest_advisory(&advisory_id, location, sha256, &tx)
             .await?;
 
-        for vuln in csaf.vulnerabilities.iter().flatten() {
-            let id = match &vuln.cve {
-                Some(cve) => cve,
-                None => continue,
-            };
+        for vuln in csaf
+            .vulnerabilities
+            .iter()
+            .flatten()
+            .filter(|e| e.cve.is_some())
+        {
+            // we filtered above, so should always be Some(_) but let's
+            // still avoid a blind unwrap.
+            if let Some(id) = &vuln.cve {
+                let advisory_vulnerability = advisory.link_to_vulnerability(id, &tx).await?;
```

Comment:
Typically, combinations of `is_some` and `unwrap` are a smell that `filter_map` should be used, e.g. `.filter_map(|e| e.cve)` but because you require both `vuln.cve` and later `vuln.product_status`, I'm guessing you need to return a tuple? To me, nested for loops are a smell, in general.